### PR TITLE
dev-python: add js2py and pyjsparser for bump of dev-python/lark-parser to 0.11.1

### PR DESCRIPTION
**Pull request by xgqt**
Move js2py and pyjsparser from ::guru to ::gentoo and become
their proxied maintainer.

This enables testing dev-python/lark-parser from
https://github.com/gentoo/gentoo/pull/19486

Thanks to telans for bringing up this PR in #gentoo-guru

Signed-off-by: Maciej Barć <xgqt@protonmail.com>

### CODE REVIEW

**Comment by xgqt** [comment]
https://github.com/gentoo/gentoo/pull/19486#discussion_r577159097

**Review by juippis — COMMENTED** [review]
Looks good but can you split adding the two packages into their own commits? Also please describe what the new package does in the git commit message. It''s usually fine to just copy the DESCRIPTION there.

**Comment by xgqt** [comment]
@juippis is this fine or should the version be also given? or should DESCRIPTION be not in the main line?

**Comment by juippis** [comment]
I'd do it:
```
dev-python/pyjsparser: new package

 * pyjsparser is a fast javascript parser based on esprima.js

Signed-off-by: ...
```
but it's fine, I can easily edit the commit message while merging. Will test tomorrow, thanks!

**Comment by xgqt** [comment]
> I'd do it:
> 
> ```
> dev-python/pyjsparser: new package
> 
>  * pyjsparser is a fast javascript parser based on esprima.js
> 
> Signed-off-by: ...
> ```
> 
> but it's fine, I can easily edit the commit message while merging. Will test tomorrow, thanks!

ok, I'll rebase

File: dev-python/pyjsparser/pyjsparser-2.7.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+DESCRIPTION="Fast javascript parser based on esprima.js"
+HOMEPAGE="
+	https://github.com/PiotrDabkowski/pyjsparser/
+	https://pypi.org/project/pyjsparser/
+"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
```
Comment (juippis) [inline dev-python/pyjsparser/pyjsparser-2.7.1.ebuild:19]: I see that at least the Github repo has test cases included. Could you perhaps make a snapshot release out of the latest commit?

https://gitweb.gentoo.org/repo/gentoo.git/tree/app-editors/kakoune/kakoune-2020.01.16_p20200601.ebuild Here's an example how to do it. Name it `pyjsparser-2.7.2_p20190422.ebuild`

File: dev-python/js2py/js2py-0.70.ebuild [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="JavaScript to Python Translator & JavaScript interpreter in Python"
+HOMEPAGE="
+	https://github.com/PiotrDabkowski/Js2Py
+	https://pypi.org/project/Js2Py/
+"
+SRC_URI="mirror://pypi/${MY_PN:0:1}/${MY_PN}/${MY_P}.tar.gz"
+
```
Comment (juippis) [inline dev-python/js2py/js2py-0.70.ebuild:19]: Same thing here.

**Review by juippis — APPROVED** [review]
Please request keywording where applicable, AFAIK these will be dependencies to a version bump for a package currently in tree? Try to get the KEYWORDS to match.

**Pull request closed without merging**